dbus_next/introspection.py

```diff
@@ -309,9 +342,30 @@ def __init__(self,
         assert_interface_name_valid(name)
 
         self.name = name
-        self.methods = methods if methods is not None else []
-        self.signals = signals if signals is not None else []
-        self.properties = properties if properties is not None else []
+        self._methods: Methods = Methods(methods)
+        self._signals: Signals = Signals(signals)
+        self._properties: Properties = Properties(properties)
```

Comment:
Changing from lists to an indexed data structure lets consumers of these data structures more easily extract the method/signal/property information they actually need, currently we seem to have to traverse each of the lists to do any useful operations on them, instead we can have methods that use a key name based lookup to get the object we actually want.